specs/experimental/alligator.md

```diff
@@ -0,0 +1,401 @@
+```
+
+This function MUST check if the `_delegator` or `_delegatee` addresses have been migrated by checking the `migrated` mapping
+from its [storage](#storage). If either address has not been migrated, the `Alligator` MUST copy the delegation
+and checkpoint data from the token contract to its own state. After copying the data, the `Alligator` MUST update
+the `migrated` mapping to reflect that the address has been migrated.
+
+When updating the subdelegation, the `subdelegateFromToken` function MUST override any previous subdelegation of the
+`_delegator` to the `_delegatee`. Afterwards, this function MUST emit a `Subdelegation` event with the given function
+parameters.
+
+#### `subdelegateBatched`
+
+Allows batch subdelegation of token voting power to multiple addresses with specified subdelegation rules. This
+function is intended to be called by users.
+
+```solidity
+function subdelegateBatched(address[] calldata _delegatees, SubdelegationRules[] calldata _rules) external
+```
+
+This function MUST check if the `msg.sender` has been migrated by checking the `migrated` mapping from its [storage](#storage).
+If it has not been migrated, the `Alligator` MUST copy the delegation and checkpoint data from the token contract
+to its own state. After copying the data, the `Alligator` MUST update the `migrated` mapping to reflect that the address
+has been migrated.
+
+This function MUST check that the length of `_delegatees` and `_rules` are equal. If the lengths are not equal, it MUST
+revert with an error.
+
+The `subdelegateBatched` function MUST iterate over each pair of `_delegatees` address and `_rules` subdelegation rule.
+At every iteration, the function MUST check the validity of the subdelegation rule, and migrate the delegatee address
+if it has not been migrated. The function MUST then update the subdelegation. Specifically for validation,
+the function MUST check that the `allowance` field of the rule does not exceed the total voting power of the delegator'
+if the `allowanceType` is `Absolute`.
+
+Afterwards, the `subdelegateBatched` function MUST emit a `Subdelegation` event with the given function parameters.
+
+#### `afterTokenTransfer`
+
+Updates the voting power of two addresses (`_from` and `_to`) after a token transfer. This function MUST only be callable
+by the `GovernanceToken` contract.
+
+```solidity
+function afterTokenTransfer(address _from, address _to, uint256 _amount) external
+```
+
+The `Alligator` MUST check if the `_from` or `_to` addresses have been migrated by checking the `migrated` mapping
+from its [storage](#storage). If either address has not been migrated, the `Alligator` MUST copy the delegation
+and checkpoint data from the token contract to its own state. After copying the data, the `Alligator` MUST update
+the `migrated` mapping to reflect that the address has been migrated.
+
+The `afterTokenTransfer` function MUST update the voting power of the `_from` and `_to` addresses with the same logic
```

Comment:
Logic will need to be updated because of the more advanced delegation logic. We should make the update logic explicit.

Reply:
updated this section in the `afterTokenTransfer` to specify that voting powers must be adjusted, and what state needs to be updated.